### PR TITLE
binary type support

### PR DESCRIPTION
**Pull request by blagoev**
Adds Binary type support

fixes https://github.com/realm/realm-dart/issues/1185

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,7 +16,8 @@
 ## 1.2.0 (2023-06-08)
 
 ### Enhancements
-* Added support for Full-Text search (simple term) queries. ([#1300](https://github.com/realm/realm-dart/pull/1300))
+  * Added support binary data type. ([#1320](https://github.com/realm/realm-dart/pull/1320))
+  * Added support for Full-Text search (simple term) queries. ([#1300](https://github.com/realm/realm-dart/pull/1300))
```
Comment (blagoev) [inline CHANGELOG.md:20]: this was just formatted

File: generator/lib/src/realm_model_info.dart [hunk 2]
```diff
@@ -34,41 +36,51 @@ class RealmModelInfo {
   Iterable<String> toCode() sync* {
     yield 'class $name extends $modelName with RealmEntity, RealmObjectBase, ${baseType.className} {';
     {
-      final allSettable = fields.where((f) => !f.type.isRealmCollection && !f.isRealmBacklink).toList();
+      final allSettableFields = fields.where((f) => !f.type.isRealmCollection && !f.isRealmBacklink).toList();
```
Comment (nielsenko) [inline generator/lib/src/realm_model_info.dart:39]: Not a fan of this name change - repeating the type in the name

**Review by nielsenko — APPROVED** [review]
Already discussed on round-table yesterday. To summarize:
- I like that we use `Uint8LIst` on the interface.
- I don't off the top of my mind have a better way to handle default values for binary.
- We could consider creating a `Matcher` that knows how to compare `Uint8List`s to avoid special cases in the tests, but I don't think it important. 
- Hopefully we can improve performance once we use the new binding generator.

File: generator/lib/src/realm_model_info.dart (hunk 2 above)
Comment (blagoev) [reply generator/lib/src/realm_model_info.dart:39]: will revert

File: generator/lib/src/realm_model_info.dart [hunk 3]
```diff
@@ -16,6 +16,8 @@
 //
 ////////////////////////////////////////////////////////////////////////////////
 
+import 'dart:typed_data';
+
```
Comment (desistefanova) [inline generator/lib/src/realm_model_info.dart:20]:
```suggestion
```

Comment (desistefanova) [reply generator/lib/src/realm_model_info.dart:20]: This is marked as unused.

**Review by desistefanova — APPROVED** [review]
It looks good to me.

**Pull request merged**